Cliqz/Extensions/BrowserViewController/BrowserViewController+ContextualMessages.swift

```diff
@@ -49,31 +53,37 @@ extension BrowserViewController {
             
             messageView.snp.makeConstraints { make in
                 make.left.right.top.equalTo(self.webViewContainer)
-                make.height.equalTo(70)
+                make.height.equalTo(contextualMessageViewHeight)
             }
             UIView.animate(withDuration: 1.0) {
                 messageView.alpha = 1
             }
             // set the ContexualMessageView so that it can be shown & hidden with the urlBar
             scrollController.contextualMessageView = contextualMessageView
-           
         }
         #endif
     }
     
-    func hideContextualMessageViewAfterDelay() {
+    func hideContextualMessageView() {
         #if PAID
-        DispatchQueue.main.asyncAfter(deadline: .now() + 20.0) {[weak self] in
-            self?.hideContextualMessageView()
+        if let messageView = self.contextualMessageView {
```

Comment:
I'd suggest to keep DispatchQueue.main.asyncAfter in this method not to rely on calling method to call it on main queue, and pass the delay as a parameter. Otherwise we have a public UI related method which potentially can be called from background thread

Reply:
DispatchQueue.main.asyncAfter  is there, the thing is git made a wrong diff. I made this method private and moved into the bottom, so git confused the changes and showing wrong diff file. You can find this method implementation with dispatch in the bottom of the class.